src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -65,22 +66,32 @@ public async Task<IActionResult> Get(string name)
         [HttpPut]
         [Route("admin/functions/{name}")]
         [Authorize(Policy = PolicyNames.AdminAuthLevel)]
-        public async Task<IActionResult> CreateOrUpdate(string name, [FromBody] FunctionMetadataResponse functionMetadata)
+        public async Task<IActionResult> CreateOrUpdate(string name, [FromBody] FunctionMetadataResponse functionMetadata, [FromServices] IEnumerable<IHostedService> hostedServices)
         {
-            if (!Utility.IsValidFunctionName(name))
+            FileMonitoringService fileMonitoringService = (FileMonitoringService)hostedServices.SingleOrDefault(x => x is FileMonitoringService);
+            try
             {
-                return BadRequest($"{name} is not a valid function name");
-            }
+                fileMonitoringService?.BlockRestartOrShutdown();
```

Comment:
As a programming model, can we shoot for the below? The service can return a disposable that will handle everything internally. This will make it cleaner and reduce mistakes.

```csharp
using (_fileMonitoringService.SuspendRestart())
{
     ...
}
```

Reply:
Yes. That was the original idea I had proposed. This method can return an `IDisposable` that we can use in this scope. Also, since this is with the file monitoring service, a better and more appropriate name would be `SuspendNotifications`

Reply:
We should also not ignore and continue here if the service is null.